test/AssetTransfer.js

```diff
@@ -1,11 +1,67 @@
 const AssetTransfer = artifacts.require('AssetTransfer');
+const truffleAssert = require('truffle-assertions');
 
 contract('AssetTransfer', (accounts) => {
-    it('should return a new instance of the contract', async () => {
-        const assetTransfer = await AssetTransfer.deployed();
-        const description = await assetTransfer.Description();
-        const price = await assetTransfer.AskingPrice();
-        assert.equal(description, 'testdescription', 'Default description not set to proper value.');
-        assert.equal(price.toNumber(), 1, 'Price was not set to the expected value of 1.');
+    let assetTransfer;
+    const owner = accounts[0];
+    const buyer = accounts[1];
+    const appraiser = accounts[2];
+    const inspector = accounts[3];
+
+    describe('Initialization', () => {
+        beforeEach('setup for Initialization', async function() {
+            assetTransfer = await AssetTransfer.deployed();
+        });
+
+        it('should return a new instance of the contract', async () => {
+            const description = await assetTransfer.Description();
+            const price = await assetTransfer.AskingPrice();
+            assert.equal(description, 'testdescription', 'Default description not set to proper value.');
+            assert.equal(price.toNumber(), 1, 'Price was not set to the expected value of 1.');
+        });
+    })
+
+    describe('MakeOffer', () => {
+        beforeEach('setup for MakeOffer', async function() {
```

Comment:
This is somehow incorrect. We should use contract instead of describe to redeploy every time. 

BeforeEach() seems not work here. Truffle only redeploys every time on contract(), not on it(). So all it() under the same contract have impact to each other.